src/seq_mv/csr_matop.c

```diff
@@ -412,24 +412,9 @@ hypre_CSRMatrixAddHost ( hypre_CSRMatrix *A,
 #endif
    {
       HYPRE_Int   ns, ne;
-      HYPRE_Int   ii, size, rest, num_threads;
       HYPRE_Int  *marker = NULL;
 
-      ii = hypre_GetThreadNum();
-      num_threads = hypre_NumActiveThreads();
-
-      size = nnzrows_A/num_threads;
```

Comment:
@victorapm Note that you have `nnzrows_A` which should be `nnzrows_C` I believe. I also use the existing function `hypre_partition1D` to simplify the code.

Reply:
Yes, right! This looks much better and solves the issue. Should we use `hypre_partition1D` at https://github.com/hypre-space/hypre/blob/cd7cd75786646c289ab15cf86337c17249e4422d/src/seq_mv/csr_matop.c#L527 ; https://github.com/hypre-space/hypre/blob/cd7cd75786646c289ab15cf86337c17249e4422d/src/seq_mv/csr_matop.c#L737 ;
https://github.com/hypre-space/hypre/blob/cd7cd75786646c289ab15cf86337c17249e4422d/src/seq_mv/csr_matop.c#L1143 ;
https://github.com/hypre-space/hypre/blob/cd7cd75786646c289ab15cf86337c17249e4422d/src/seq_mv/csr_matop.c#L1372 ;

as well? Thanks!

Reply:
Yes, I prefer using that function, but what's more important is we don't have similar bugs at other places. Can you please double check all of them? Thanks!

Reply:
Yes! I just double checked and they are correct. Thanks!

Reply:
> Yes! I just double checked and they are correct. Thanks!

Great! Do you also want to make more changes using `hypre_partition1D`? I am fine with it or without. Thanks!

Reply:
I think it's a good, idea. Then we reduce code duplication. Can i do this and commit to this branch?

Reply:
> I think it's a good idea. Then we reduce code duplication. Can i do this and commit to this branch?

Of course you can :)